ui/src/app/analytics/analytics.service.ts

```diff
@@ -138,4 +161,119 @@ export class AnalyticsService {
     this.counters.update(page);
     return page;
   }
+
+  addNewDashboardItem(index?: number): DashboardItem {
+    if (!this.dashboardItems) {
+      this.dashboardItems = [];
+    }
+    const dashboardItem = new DashboardItem();
+    dashboardItem.id = this.rowId++;
+    dashboardItem.refreshRate = 2;
+    dashboardItem.visualization = '';
+
+    console.log('>>>', this.dashboardItems);
+    if (index) {
+      this.dashboardItems.splice(index, 0, dashboardItem);
+    } else {
+      this.dashboardItems.push(dashboardItem);
+    }
+    console.log('>>>', this.dashboardItems);
+    return dashboardItem;
+  }
+
+  removeDashboardItem(index: number): void {
+    if (!this.dashboardItems || this.dashboardItems.length === 0) {
+      return;
+    }
+    this.dashboardItems.splice(index, 1);
+  }
+
+  getAllDashboardItems(): Observable<DashboardItem> {
+    this.addNewDashboardItem();
```

Comment:
this line causes new dashboard items to be added when leaving and entering the page.